controllers/component_build_controller_unit_test.go

```diff
@@ -1641,6 +1641,11 @@ func TestGetGitProvider(t *testing.T) {
 			componentRepoUrl: "12345",
 			expectError:      true,
 		},
+		{
+			name:             "should return error if git source URL is empty",
+			componentRepoUrl: "",
+			expectError:      true,
+		},
```

Comment:
For my understanding, this test case is not related to the null pointer exception right?

Reply:
Yes, right